crates/l2/proposer/mod.rs

```diff
@@ -63,7 +54,28 @@ impl Proposer {
         })
     }
 
-    pub async fn start(&self, head_block_hash: H256) -> Result<(), ProposerError> {
+    pub async fn run(&self, store: Store) {
+        loop {
+            let head_block_hash = {
+                let current_block_number = store
+                    .get_latest_block_number()
+                    .expect("store.get_latest_block_number")
```

Comment:
Remove these expects and the ones below

Reply:
These requests should also be part of the main logic

Reply:
fixed